### PR TITLE
Doc Revamp

### PR DESCRIPTION
**Pull request by msaroufim**
We have a leaderboard now, we have a finetuning example, there's lots of important rules on discord that aren't here so this is fixes everything

Also updated the leaderboard page on our site https://github.com/llm-efficiency-challenge/llm-efficiency-challenge.github.io/blob/main/leaderboard.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
 
 Your `Dockerfile` will expose a simple HTTP server, which needs to implement 2 endpoints `/process` and `/tokenize`. We will build that `Dockerfile` and expect it to launch an HTTP server. Once that server is launched, we will make requests to it via HELM and record your results.
 
+At a high level the flow you should follow to ensure a strong submission is
```
Comment (lantiga) [inline README.md]: Sentence sounds incomplete?

File: README.md [hunk 2]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+## Submission
+
+The submission in this repository is a basic implementation of the setting up an HTTP server in accordance to the `open_api` spec. It includes a sample solution built off of [Lit-GPT](https://github.com/Lightning-AI/lit-gpt) and open-llama weights that participants can reference or modify as they see fit.
 
 You can use the provided code as a reference or starting point for your own implementation. The `main.py` file contains the simple FastAPI server, and you can modify it to suit your needs.
 
-You can find the toy submission [here](/toy-submission).
+You can find the Lit-GPT submission [here](sample-submissions/lit-gpt/) with instructions on how to run it locally.
+
+## Eval locally HELM
```
Comment (lantiga) [inline README.md]:
```suggestion
## Eval HELM locally
```

Comment (rasbt) [inline README.md]: Perhaps "Evaluate Your Model Locally Using HELM"?

File: README.md [hunk 3]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+
+## Finetune
 
-### OpenAPI Specification
+It's likely that an untuned base model won't give you satisfactory results, in that case you might find it helpful to do some additional finetuning. There are many frameworks to do this but we've created 2 sample submissions to do so
+1. [lit-gpt](/sample-submissions/lit-gpt/)
+2. [llama-recipes](/sample-submissions/llama_recipes/)
+
+
+### Your own submission template
```
Comment (rasbt) [inline README.md]: Perhaps "Create Your Own Submission Template"

File: README.md [hunk 4]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+
+Note that we've offered 2 sample submissions, our evaluation infrastructure is generic and only assumes an HTTP client so you can use a finetuning framework in python like the ones we've suggested but also any non based python framework you like using.
```
Comment (rasbt) [inline README.md]: python -> Python

File: README.md [hunk 5]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
 
-As you're working on your submission `Dockerfile` you'll want to test it out locally to make sure your contribution works as expected before you submit it.
+You can interact with it by DM'ing it with a zipped file of your sample submission and message it to either `eval A100` or `eval 4090`. More details on the bot are [here](https://discord.com/channels/1124130156336922665/1124134272631054447/1151718598818156645)
 
-To learn more about how to test your submission with HELM, please follow the instructions [here](helm.md).
+Once you make a submission the bot will inform you whether your submission failed or succeeded and after a few hours will publicly post your results. Your submission will remain private to other competitors.
+
+The end to end flow is described [here](leaderboard.md)
+
+## Final Eval Submission
```
Comment (rasbt) [inline README.md]: - Final Eval Submission -> Final Leaderboard Submission

- please sure -> please make sure

Comment (rasbt) [reply README.md]: Perhaps also add the deadline

File: README.md [hunk 6]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+
+When you registered for the competition you would have needed to create a github repo with your `Dockerfile`, in case the location is ambiguous please sure to let us know in your `README.md`. The organizers will take your `Dockerfile` and run it as is and compute a baseline eval score. The purpose of this step is to primarily filter out broken submissions or submissions that can't outperform the unfinetuned sample submissions.
+
+
+## Final Finetune Submission
```
Comment (rasbt) [inline README.md]: How about "Evaluating the Final Submission"

File: README.md [hunk 7]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
 
 Your `Dockerfile` will expose a simple HTTP server, which needs to implement 2 endpoints `/process` and `/tokenize`. We will build that `Dockerfile` and expect it to launch an HTTP server. Once that server is launched, we will make requests to it via HELM and record your results.
 
+At a high level the flow you should follow to ensure a strong submission is
+1. Pick approved LLMs and datasets from [here](https://llm-efficiency-challenge.github.io/challenge)
+2. Start with one of [sample-submissions](sample-submissions) and make sure it runs
+3. Evaluate it locally on your own A100 or 4090, if you don't have funding for either please see the [GPU funding](#gpu-funding) section for some more options
```
Comment (rasbt) [inline README.md]: 40Gb A100?

File: README.md [hunk 8]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
 
-The submission in this repository is a basic implementation of the setting up an HTTP server in accordance to the open_api spec. It includes a sample solution built off of [Lit-GPT](https://github.com/Lightning-AI/lit-gpt) and open-llama weights that participants can reference or modify as they see fit.
+The LLM space has complex licenses which can make it difficult to figure out what's permittable to use in a competition to streamline this process we've shortlisted a few models and datasets we know are safe to use [here](https://llm-efficiency-challenge.github.io/challenge)
 
-### Usage
+That said the LLM space is fast moving so if you'd like to use a dataset or model that isn't on our list make sure to ask us about it on [https://discord.gg/XJwQ5ddMK7](https://discord.gg/XJwQ5ddMK7)
+
+## Submission
+
+The submission in this repository is a basic implementation of the setting up an HTTP server in accordance to the `open_api` spec. It includes a sample solution built off of [Lit-GPT](https://github.com/Lightning-AI/lit-gpt) and open-llama weights that participants can reference or modify as they see fit.
```
Comment (lantiga) [inline README.md:37]: I would make sure people understand that creating one (and only one) Dockerfile is a requirement.
Also that you'll need to host your weights somewhere that will be downloaded either during docker build or at runtime (i.e. not in the repo / zip file).

File: README.md [hunk 9]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+You can interact with it by DM'ing it with a zipped file of your sample submission and message it to either `eval A100` or `eval 4090`. More details on the bot are [here](https://discord.com/channels/1124130156336922665/1124134272631054447/1151718598818156645)
 
-To learn more about how to test your submission with HELM, please follow the instructions [here](helm.md).
+Once you make a submission the bot will inform you whether your submission failed or succeeded and after a few hours will publicly post your results. Your submission will remain private to other competitors.
+
+The end to end flow is described [here](leaderboard.md)
+
+## Final Eval Submission
+
+When you registered for the competition you would have needed to create a github repo with your `Dockerfile`, in case the location is ambiguous please sure to let us know in your `README.md`. The organizers will take your `Dockerfile` and run it as is and compute a baseline eval score. The purpose of this step is to primarily filter out broken submissions or submissions that can't outperform the unfinetuned sample submissions.
```
Comment (rasbt) [inline README.md]: How do people register, is it via https://llm-efficiency-challenge.github.io/submission? 

If yes, why is it called Submission and not Registration?

**Review by rasbt — COMMENTED** [review]
That looks great, Mark. Here are just a few small comments

File: README.md [hunk 10]
```diff
@@ -6,33 +6,87 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
+The [Lightning.ai](https://lightning.ai/) has built a Discord based for us. You can find it on discord by its name `evalbot#4372`.
 
-As you're working on your submission `Dockerfile` you'll want to test it out locally to make sure your contribution works as expected before you submit it.
+You can interact with it by DM'ing it with a zipped file of your sample submission and message it to either `eval A100` or `eval 4090`. More details on the bot are [here](https://discord.com/channels/1124130156336922665/1124134272631054447/1151718598818156645)
 
-To learn more about how to test your submission with HELM, please follow the instructions [here](helm.md).
+Once you make a submission the bot will inform you whether your submission failed or succeeded and after a few hours will publicly post your results. Your submission will remain private to other competitors.
```
Comment (lantiga) [inline README.md]: Maybe add how long one should expect evaluation to take (especially if you're queued up). And also  how important it is to first verify that the evaluation succeeds locally before submitting.

File: leaderboard.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
+# Leaderboard usage 
+
+
+## How to use the leaderboard
+The Lightning.ai team @lntg has built us a leaderboard on Discord. This is the single best way you can make sure your submissions actually work before the submission, try to beat the unfinetuned toy submission as a starting point.
```
Comment (lantiga) [inline leaderboard.md]: lntg id my username :-) (I realize now how close it is to be a contraction of lightning, but it's actually a contraction of lantiga)

File: README.md (hunk 9 above)
Comment (msaroufim) [reply README.md]: It is unfortunately both lemme make this clearer

File: leaderboard.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+You might have noticed a new friendly bot has joined the server called @evalbot  to use it
+1. DM the bot with eval 4090 or eval A100 with a zipped file of your submission (You can also just openly message the bot but DM'ing will protect your secret sauce)
```
Comment (lantiga) [inline leaderboard.md]:
```suggestion
1. DM the bot `eval 4090` or `eval A100` and attach a zipped file of your submission to the message (You can also just openly message the bot but DM'ing will protect your secret sauce)
```

File: leaderboard.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+2. If successful the bot will give you a job ID and a running status, the eval will take roughly 1-2h so be patient if you're top of queue
+3. Once the bot completes your run it will update either the ⁠leaderboard_4090  or ⁠leaderboard_a100 channel, we will not be monitoring these 2 text channels they will be purely for the bot to post the new updated leaderboard
```
Comment (lantiga) [inline leaderboard.md:10]: Add expectations about timing.

**Review by lantiga — APPROVED** [review]
A few suggestions

File: leaderboard.md (hunk 11 above)
Comment (msaroufim) [reply leaderboard.md]: you landed in the right place it seems haha

File: README.md [hunk 14]
```diff
@@ -6,33 +6,94 @@ At a high level the key thing you will contribute is a `Dockerfile`, which will
...
-The submission in this repository is a basic implementation of the setting up an HTTP server in accordance to the open_api spec. It includes a sample solution built off of [Lit-GPT](https://github.com/Lightning-AI/lit-gpt) and open-llama weights that participants can reference or modify as they see fit.
+## Submission
 
-### Usage
+The submission in this repository is a basic implementation of the setting up an HTTP server in accordance to the `open_api` spec. It includes a sample solution built off of [Lit-GPT](https://github.com/Lightning-AI/lit-gpt) and open-llama weights that participants can reference or modify as they see fit.
 
 You can use the provided code as a reference or starting point for your own implementation. The `main.py` file contains the simple FastAPI server, and you can modify it to suit your needs.
 
-You can find the toy submission [here](/toy-submission).
+You can find the Lit-GPT submission [here](sample-submissions/lit-gpt/) with instructions on how to run it locally.
```
Comment (chauhang) [inline README.md]: Will be good to also mention llama-recipes submission here

**Review by chauhang — CHANGES_REQUESTED** [review]
Changes look good overall. Please see inline

File: README.md (hunk 6 above)
Comment (weiweiy) [reply README.md]: Looks good to me :)

**Pull request merged**